ksqldb-examples/src/main/java/io/confluent/ksql/datagen/ProducerFactory.java

```diff
@@ -50,8 +54,18 @@ static DataGenProducer getProducer(
     final SerializerFactory<Struct> keySerializerFactory =
         keySerializerFactory(keyFormat, ksqlConfig, srClient);
 
+    if (valueDelimiter != null && !(valueFormat instanceof DelimitedFormat)) {
+      throw new IllegalArgumentException(
+          "valueDelimiter can only be specified with delimited format");
+    }
```

Comment:
nite: consider moving this into the if block below that already checks `valueDelimiter != null`

Reply:
ack